framework/web/JsonResponseFormatter.php

```diff
@@ -117,6 +117,8 @@ protected function formatJson($response)
                 $options |= JSON_PRETTY_PRINT;
             }
             $response->content = Json::encode($response->data, $options);
+        } else {
+            $response->content = 'null';
```

Comment:
content should not be overwritten, if data is null, your change contradicts the documentation...

Reply:
this seems to be fixed by https://github.com/yiisoft/yii2/commit/3c091b802a03730c3591eae1d51cceebf88aadde